Sources/Hummingbird/Middleware/MiddlewareModule/MiddlewareFixedTypeBuilder.swift

```diff
@@ -64,4 +36,28 @@ public enum MiddlewareFixedTypeBuilder<Input, Output, Context> {
     ) -> _Middleware2<M0, M1> where M0.Input == M1.Input, M0.Output == M1.Output, M0.Context == M1.Context {
         _Middleware2(m0, m1)
     }
+
+    public static func buildOptional<M0: MiddlewareProtocol>(_ component: M0?) -> _ConditionalMiddleware<M0, _EmptyMiddleware<M0>> {
```

Comment:
If `_ConditionalMiddleware` isn't used by `buildEither` then this is the only place it is used. Can we instead just write an `OptionalMiddleware` and get rid of `_ConditionMiddleware` and `_EmptyMiddleware`.

Reply:
Good feedback! 👍 I've pushed changes to address the recommendations.